Doc/library/sqlite3.rst

```diff
@@ -2269,8 +2339,67 @@ Explanation
 Transaction control
 ^^^^^^^^^^^^^^^^^^^
 
-The :mod:`!sqlite3` module does not adhere to the transaction handling recommended
-by :pep:`249`.
+.. _sqlite3-transaction-control-autocommit:
+
+Transaction control via the ``autocommit`` attribute
+""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+The recommended way of controlling transaction behaviour is through
+the :attr:`Connection.autocommit` attribute.
+This attribute should preferrably be set using the *autocommit* parameter
+of :func:`connect`.
+
+It is recommended to set *autocommit* to ``False``,
+implying :pep:`249`-compliant transaction control.
+This means:
+
+* :mod:`!sqlite3` ensures that a transaction is always open,
+  so :meth:`~Connection.commit` and :meth:`~Connection.rollback`
+  will implicitly open a new transaction immediately after closing
+  the pending transaction.
+  :mod:`!sqlite3` uses ``BEGIN DEFERRED`` statements when opening transactions.
+* Transactions should be committed explicitly
+  using :meth:`!commit`.
+* Transactions should be rolled back explicitly
+  using :meth:`!rollback`.
+* An implicit rollback is performed if the database is
+  :meth:`~Connection.close`-ed with pending changes.
+
+Set *autocommit* to ``True`` to enable SQLite's `autocommit mode`_.
+In this mode, :meth:`~Connection.commit()` and :meth:`~Connection.rollback()`
+have no effect.
+
+Set *autocommit* to :const:`COMPAT_TRANSACTIONAL_CONTROL`
+to leave transaction control behaviour to the
+:attr:`Connection.isolation_level` attribute.
+See :ref:`sqlite3-transaction-control-isolation-level` for more information.
+
+.. note::
+
+   :attr:`~Connection.autocommit` and SQLite's `autocommit mode`_
+   are not linked.
+   For example, explicitly issuing a ``BEGIN`` statement when
+   :attr:`!autocommit` is ``True`` will not change the value of
+   the *autocommit* attribute.
+
+   Use :attr:`Connection.in_transaction` to query the low-level SQLite
+   autocommit mode.
```

Comment:
You might want to move this note to right below the description of the `True` value, where it both seems more directly relevant, and also looks a little better/clearer (not clashing with the other note and making the start of the next section less obvious).

Reply:
Hmm, IMO the note under the `autocommit` attribute avoids the What If ish parts and simply clearly states the facts, so maybe either move that one here, or remove this one and just leave that one?

Reply:
Just stating the facts is almost always a good thing. I'm removing the _what if_.